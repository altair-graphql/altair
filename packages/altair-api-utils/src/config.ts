export type ClientEnvironment = 'development' | 'production' | 'testing';
export interface APIClientOptions {
  apiBaseUrl: string;
  loginClientUrl: string;
  sandboxUrl: string;
}
export const getClientConfig = (
  env: ClientEnvironment = 'development'
): APIClientOptions => {
  switch (env) {
    case 'production':
      return {
        apiBaseUrl:
          process.env.ALTAIR_API_CLIENT_BASE_URL ?? 'https://api.altairgraphql.dev',
        loginClientUrl:
          process.env.ALTAIR_API_CLIENT_LOGIN_CLIENT_URL ??
          'https://redir.altairgraphql.dev',
        sandboxUrl:
          process.env.ALTAIR_API_CLIENT_SANDBOX_URL ??
          'https://sandbox.altairgraphql.dev',
      };
  }

  return {
    apiBaseUrl: 'http://localhost:3000',
    loginClientUrl: 'http://localhost:1234',
    sandboxUrl: 'http://localhost:5123',
  };
};
