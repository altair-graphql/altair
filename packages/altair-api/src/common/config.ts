export type AiModelProvider = 'anthropic' | 'openai' | 'ollama';

const config = {
  nest: {
    port: 3000,
  },
  cors: {
    enabled: true,
  },
  swagger: {
    enabled: true,
    title: 'Altair API',
    description: 'The Altair API description',
    version: '1.0',
    path: 'swagger',
  },
  security: {
    shortExpiresIn: '30s',
    expiresIn: '2d',
    refreshIn: '7d',
    bcryptSaltOrRound: 10,
  },
  ai: {
    modelProvider: process.env.AI_MODEL_PROVIDER as AiModelProvider | undefined,
    openai: {
      apiKey: process.env.OPENAI_API_KEY,
      // https://platform.openai.com/docs/models/overview
      model: process.env.OPENAI_MODEL_NAME ?? 'gpt-3.5-turbo-0125',
    },
    ollama: {
      baseUrl: process.env.OLLAMA_BASE_URL,
      // https://ollama.com/library
      model: process.env.OLLAMA_MODEL_NAME ?? 'llama3',
    },
    anthropic: {
      apiKey: process.env.ANTHROPIC_API_KEY,
      // https://docs.anthropic.com/en/docs/about-claude/models#model-names
      model: process.env.ANTHROPIC_MODEL_NAME ?? 'claude-3-haiku-20240307',
    },
  },
};

export type Config = typeof config;
export type NestConfig = Config['nest'];
export type SwaggerConfig = Config['swagger'];
export type SecurityConfig = Config['security'];
export type CorsConfig = Config['cors'];

export default () => config;
