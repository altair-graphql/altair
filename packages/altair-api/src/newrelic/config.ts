/**
 * New Relic agent configuration.
 *
 * See lib/config/default.js in the agent distribution for a more complete
 * description of configuration variables and their potential values.
 */
exports.config = {
  /**
   * Array of application names.
   */
  app_name: [process.env.NEW_RELIC_APP_NAME],
  /**
   * Your New Relic license key.
   */
  license_key: process.env.NEW_RELIC_LICENSE_KEY,
  /**
   * This setting controls distributed tracing.
   * Distributed tracing lets you see the path that a request takes through your
   * distributed system. Enabling distributed tracing changes the behavior of some
   * New Relic features, so carefully consult the transition guide before you enable
   * this feature: https://docs.newrelic.com/docs/transition-guide-distributed-tracing
   * Default is true.
   */
  distributed_tracing: {
    /**
     * Enables/disables distributed tracing.
     *
     * @env NEW_RELIC_DISTRIBUTED_TRACING_ENABLED
     */
    enabled: true,
  },
  logging: {
    /**
     * Level at which to log. 'trace' is most useful to New Relic when diagnosing
     * issues with the agent, 'info' and higher will impose the least overhead on
     * production applications.
     */
    level: 'trace',
  },
  /**
   * When true, all request headers except for those listed in attributes.exclude
   * will be captured for all traces, unless otherwise specified in a destination's
   * attributes include/exclude lists.
   */
  allow_all_headers: true,
  attributes: {
    /**
     * Prefix of attributes to exclude from all destinations. Allows * as wildcard
     * at end.
     *
     * NOTE: If excluding headers, they must be in camelCase form to be filtered.
     *
     * @env NEW_RELIC_ATTRIBUTES_EXCLUDE
     */
    exclude: [
      'request.headers.cookie',
      'request.headers.authorization',
      'request.headers.proxyAuthorization',
      'request.headers.setCookie*',
      'request.headers.x*',
      'response.headers.cookie',
      'response.headers.authorization',
      'response.headers.proxyAuthorization',
      'response.headers.setCookie*',
      'response.headers.x*',
    ],
  },
};
