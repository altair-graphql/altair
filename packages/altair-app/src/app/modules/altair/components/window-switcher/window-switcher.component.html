<ul
  cdkDropList
  cdkDropListOrientation="horizontal"
  class="window-switcher__list"
  [ngClass]="{'window-switcher__no-scrollbar': !enableScrollbar}"
  [cdkDropListData]="windowIds"
  (cdkDropListDropped)="onDropEnd($event)"
>
  <li
    cdkDrag
    class="window-switcher__item"
    [ngClass]="{ 'window-switcher__item--active': windowId === activeWindowId }"
    *ngFor="let windowId of windowIds; let index = index"
    [contextMenu]="windowTabMenu"
    [contextMenuSubject]="{ windowId: windowId, wTitle: wTitle, index: index }"
    (click)="onClickWindow(windowId)"
    (dblclick)="editWindowNameInput(windowId, wTitle)"
  >
    <app-icon
      *ngIf="windows[windowId]?.layout?.windowIdInCollection"
      name="folder"
      nz-tooltip
      [nzTooltipTitle]="'COLLECTION_OPEN_QUERY_TOOLTIP' | translate"
      nzTooltipPlacement="bottom"
    ></app-icon>
    <div
      #wTitle
      class="window-switcher__input"
      [ngClass]="{
        'window-switcher__input--editing': windowId === windowIdEditing
      }"
      [attr.contenteditable]="windowId === windowIdEditing"
      (blur)="saveWindowName(windowId, wTitle.innerText)"
      (keydown.enter)="saveWindowName(windowId, wTitle.innerText)"
    >
      {{ windows[windowId]?.layout?.title || '' }}
    </div>
    <div
      track-id="close_window_tab"
      class="window-switcher__close"
      *ngIf="windowIds.length > 1"
      (click)="closeWindow(windowId)"
    >
      &#x00D7;
    </div>
  </li>
  <li
    class="window-switcher__item"
    (click)="newWindowChange.next($event)"
    *ngIf="windowIds.length < maxWindowCount"
  >
    {{ 'ADD_NEW_WINDOW_TEXT' | translate }}
  </li>
</ul>

<context-menu>
  <ng-template
    contextMenuItem
    (execute)="editWindowNameInput($event.item.windowId, $event.item.wTitle)"
  >
    <span class="menu-icon">
      <app-icon name="edit"></app-icon>
    </span>
    <span class="menu-text">
      {{ 'EDIT_WINDOW_TEXT' | translate }}
    </span>
  </ng-template>
  <ng-template
    contextMenuItem
    (execute)="duplicateWindow($event.item.windowId)"
  >
    <span class="menu-icon">
      <app-icon name="copy"></app-icon>
    </span>
    <span class="menu-text">
      {{ 'DUPLICATE_WINDOW_TEXT' | translate }}
    </span>
  </ng-template>
  <ng-template contextMenuItem [divider]="true"></ng-template>
  <ng-template contextMenuItem (execute)="closeWindow($event.item.windowId)">
    <span class="menu-icon">
      <app-icon name="x-square"></app-icon>
    </span>
    <span class="menu-text">
      {{ 'CLOSE_WINDOW_TEXT' | translate }}
    </span>
  </ng-template>
  <ng-template
    contextMenuItem
    (execute)="closeWindowsToTheRight($event.item.index)"
  >
    <span class="menu-icon">
      <app-icon name="circle"></app-icon>
    </span>
    <span class="menu-text">
      {{ 'CLOSE_WINDOWS_TO_THE_RIGHT_TEXT' | translate }}
    </span>
  </ng-template>
  <ng-template
    contextMenuItem
    (execute)="closeOtherWindows($event.item.windowId)"
  >
    <span class="menu-icon">
      <app-icon name="circle"></app-icon>
    </span>
    <span class="menu-text">
      {{ 'CLOSE_OTHER_WINDOWS_TEXT' | translate }}
    </span>
  </ng-template>
  <ng-template
    *ngIf="closedWindows?.length"
    contextMenuItem
    (execute)="reopenClosedTab()"
  >
    <span class="menu-icon">
      <app-icon name="plus-square"></app-icon>
    </span>
    <span class="menu-text">
      {{ 'REOPEN_CLOSED_WINDOW_TEXT' | translate }}
    </span>
  </ng-template>
</context-menu>
