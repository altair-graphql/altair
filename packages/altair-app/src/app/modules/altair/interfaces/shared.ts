export interface IDictionary<V = any> {
  [key: string]: V;
}

export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;

export type TODO = any;

export type UnknownError = TODO;

export interface TrackByIdItem {
  id: string;
}
