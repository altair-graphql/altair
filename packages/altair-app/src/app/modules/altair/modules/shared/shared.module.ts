import { NgModule, ModuleWithProviders } from '@angular/core';
import { CommonModule } from '@angular/common';

import { DragDropModule } from '@angular/cdk/drag-drop';
import { TranslateModule } from '@ngx-translate/core';
import { NgPipesModule } from 'ngx-pipes';
import { NguiAutoCompleteModule } from '@ngui/auto-complete';
import { MarkdownModule } from 'ngx-markdown';
import { HotToastModule } from '@ngneat/hot-toast';
import { ResizableModule } from 'angular-resizable-element';

import { NzButtonModule } from 'ng-zorro-antd/button';
import { NzFormModule } from 'ng-zorro-antd/form';
import { NzSelectModule } from 'ng-zorro-antd/select';
import { NzModalModule } from 'ng-zorro-antd/modal';
import { NzDropDownModule } from 'ng-zorro-antd/dropdown';
import { NzSwitchModule } from 'ng-zorro-antd/switch';
import { NzListModule } from 'ng-zorro-antd/list';
import { NzRadioModule } from 'ng-zorro-antd/radio';
import { NzAlertModule } from 'ng-zorro-antd/alert';
import { NzLayoutModule } from 'ng-zorro-antd/layout';
import { NzMenuModule } from 'ng-zorro-antd/menu';
import { NzCollapseModule } from 'ng-zorro-antd/collapse';
import { NzInputModule } from 'ng-zorro-antd/input';
import { NzInputNumberModule } from 'ng-zorro-antd/input-number';
import { NzAutocompleteModule } from 'ng-zorro-antd/auto-complete';
import { NzCheckboxModule } from 'ng-zorro-antd/checkbox';
import { NzToolTipModule } from 'ng-zorro-antd/tooltip';
import { NzTabsModule } from 'ng-zorro-antd/tabs';
import { NzTableModule } from 'ng-zorro-antd/table';
import { NzTreeSelectModule } from 'ng-zorro-antd/tree-select';
import { NzGridModule } from 'ng-zorro-antd/grid';
import { NzPopoverModule } from 'ng-zorro-antd/popover';

import { IconsModule } from '../icons/icons.module';
import { NzAvatarModule } from 'ng-zorro-antd/avatar';

const AntdComponentModules = [
  NzButtonModule,
  NzInputModule,
  NzInputNumberModule,
  NzAutocompleteModule,
  NzCheckboxModule,
  NzFormModule,
  NzGridModule,
  NzTabsModule,
  NzSelectModule,
  NzModalModule,
  NzDropDownModule,
  NzSwitchModule,
  NzListModule,
  NzRadioModule,
  NzAlertModule,
  NzLayoutModule,
  NzMenuModule,
  NzCollapseModule,
  NzToolTipModule,
  NzTableModule,
  NzTreeSelectModule,
  NzPopoverModule,
  NzAvatarModule,
  // NgZorroAntdModule,
];

@NgModule({
  imports: [
    CommonModule,
    DragDropModule,
    TranslateModule.forRoot(),
    NgPipesModule,
    NguiAutoCompleteModule, // use antd
    MarkdownModule.forRoot(),
    ResizableModule,
    IconsModule,
    // ToastrModule.forRoot({
    //   newestOnTop: false,
    //   closeButton: true,
    //   positionClass: 'toast-top-center',
    //   enableHtml: true,
    //   countDuplicates: true,
    //   preventDuplicates: true,
    //   resetTimeoutOnDuplicate: true,
    // }),
    // HotToastModule.forRoot({
    //   position: 'top-center',
    //   reverseOrder: true,
    //   dismissible: true,
    //   closeStyle: {
    //     backgroundImage: `none`,
    //   },
    // }),
    ...AntdComponentModules,
  ],

  // DON'T FORGET TO EXPORT MODULE
  exports: [
    DragDropModule,
    TranslateModule,
    NgPipesModule,
    NguiAutoCompleteModule,
    MarkdownModule,
    ResizableModule,
    IconsModule,
    HotToastModule,
    ...AntdComponentModules,
  ],
})
export class SharedModule {
  static forRoot(): ModuleWithProviders<SharedModule> {
    return {
      ngModule: SharedModule,
      providers: [],
    };
  }
}
