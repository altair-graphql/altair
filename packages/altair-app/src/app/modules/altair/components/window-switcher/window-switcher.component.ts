import { CdkDragDrop } from '@angular/cdk/drag-drop';
import {
  Component,
  Input,
  Output,
  EventEmitter,
  HostBinding,
} from '@angular/core';
import { AltairConfig } from 'altair-graphql-core/build/config';
import { PerWindowState } from 'altair-graphql-core/build/types/state/per-window.interfaces';
import { WindowState } from 'altair-graphql-core/build/types/state/window.interfaces';
import {
  NzContextMenuService,
  NzDropdownMenuComponent,
} from 'ng-zorro-antd/dropdown';

import { debug } from '../../utils/logger';

@Component({
  selector: 'app-window-switcher',
  templateUrl: './window-switcher.component.html',
  styleUrls: ['./window-switcher.component.scss'],
})
export class WindowSwitcherComponent {
  @Input() windows: WindowState = {};
  @Input() windowIds: string[] = [];
  @Input() closedWindows: PerWindowState[] = [];
  @Input() activeWindowId = '';
  @Input() isElectron = false;
  @Input() enableScrollbar = false;
  @Output() activeWindowChange = new EventEmitter();
  @Output() newWindowChange = new EventEmitter();
  @Output() removeWindowChange = new EventEmitter();
  @Output() duplicateWindowChange = new EventEmitter();
  @Output() windowNameChange = new EventEmitter();
  @Output() repositionWindowChange = new EventEmitter();
  @Output() reopenClosedWindowChange = new EventEmitter();

  @HostBinding('class.window-switcher__no-scrollbar') get noScrollbar() {
    return !this.enableScrollbar;
  }

  windowIdEditing = '';
  maxWindowCount = this.altairConfig.max_windows;

  constructor(
    private altairConfig: AltairConfig,
    private nzContextMenuService: NzContextMenuService
  ) {}

  onDropEnd(event: CdkDragDrop<any, any, any>) {
    this.moveWindow(event.previousIndex || 0, event.currentIndex || 0);
  }

  onClickWindow(windowId: string) {
    this.activeWindowChange.next(windowId);
  }

  editWindowNameInput(windowId: string, wTitle: HTMLElement) {
    this.windowIdEditing = windowId;
    setTimeout(() => wTitle.focus(), 0);
  }

  saveWindowName(windowId: string, windowName: string) {
    if (this.windowIdEditing) {
      this.windowNameChange.next({ windowId, windowName });
      this.windowIdEditing = '';
    }
  }

  moveWindow(currentPosition: number, newPosition: number) {
    this.repositionWindowChange.next({ currentPosition, newPosition });
  }

  closeWindow(windowId: string) {
    return this.removeWindowChange.next(windowId);
  }

  closeWindowsToTheRight(curIndex: number) {
    const lowerBound = curIndex + 1;
    if (lowerBound >= this.windowIds.length) {
      return;
    }
    return this.windowIds
      .filter((wid, i) => i > curIndex)
      .map((_) => this.closeWindow(_));
  }

  closeOtherWindows(windowId: string) {
    return this.windowIds
      .filter((wid) => wid !== windowId)
      .map((_) => this.closeWindow(_));
  }

  duplicateWindow(windowId: string) {
    this.duplicateWindowChange.next(windowId);
  }

  reopenClosedTab() {
    this.reopenClosedWindowChange.emit();
  }

  contextMenu($event: MouseEvent, menu: NzDropdownMenuComponent): void {
    this.nzContextMenuService.create($event, menu);
  }

  closeMenu(): void {
    this.nzContextMenuService.close();
  }

  log(str: string) {
    debug.log(str);
  }
}
