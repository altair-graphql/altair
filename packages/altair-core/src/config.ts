import { SubscriptionProviderIds, WEBSOCKET_PROVIDER_ID } from './subscriptions';
import { IDictionary } from './types/shared';
import { IInitialEnvironments } from './types/state/environments.interfaces';
import { HttpVerb } from './types/state/query.interfaces';
import { SettingsState } from './types/state/settings.interfaces';
import isElectron from './utils/is_electron';

const isTranslateMode = (window as any).__ALTAIR_TRANSLATE__;

export interface AltairWindowOptions {

  /**
   * Initial name of the window
   */
  initialName?: string;

  /**
   * URL to set as the server endpoint
   */
   endpointURL?: string;

  /**
  * URL to set as the subscription endpoint. This can be relative or absolute.
  */
  subscriptionsEndpoint?: string;

  /**
   * URL protocol for the subscription endpoint. This is used if the specified subscriptions endpoint is relative.
   * e.g. wss
   */
  subscriptionsProtocol?: string;

  /**
  * Initial query to be added
  */
  initialQuery?: string;

  /**
  * Initial variables to be added
  */
  initialVariables?: string;

  /**
  * Initial pre-request script to be added
  */
  initialPreRequestScript?: string;

  /**
  * Initial post-request script to be added
  */
  initialPostRequestScript?: string;

  /**
  * Initial headers object to be added
  * @example
  * {
  *  'X-GraphQL-Token': 'asd7-237s-2bdk-nsdk4'
  * }
  */
  initialHeaders?: IDictionary;

  /**
   * Initial subscriptions provider
   *
   * @default "websocket"
   */
  initialSubscriptionsProvider?: SubscriptionProviderIds;

  /**
   * Initial subscriptions connection params
   */
  initialSubscriptionsPayload?: IDictionary;

  /**
   * HTTP method to use for making requests
   */
  initialHttpMethod?: HttpVerb;
}

export interface AltairConfigOptions extends AltairWindowOptions {

  /**
   * Initial Environments to be added
   * @example
   * {
   *   base: {
   *     title: 'Environment',
   *     variables: {}
   *   },
   *   subEnvironments: [
   *     {
   *       title: 'sub-1',
   *       variables: {}
   *     }
   *   ]
   * }
   */
  initialEnvironments?: IInitialEnvironments;

  /**
   * Namespace for storing the data for the altair instance.
   * Use this when you have multiple altair instances running on the same domain.
   * @example
   * instanceStorageNamespace: 'altair_dev_'
   */
  instanceStorageNamespace?: string;

  /**
   * Initial app settings to use
   */
  initialSettings?: Partial<SettingsState>;

  /**
   * Indicates if the state should be preserved for subsequent app loads
   *
   * @default true
   */
  preserveState?: boolean;

  /**
   * List of options for windows to be loaded
   */
  initialWindows?: AltairWindowOptions[];
}

export class AltairConfig {
  donation = {
    url: 'https://opencollective.com/altair/donate',
    action_count_threshold: 50
  };
  ga = 'UA-41432833-6';
  add_query_depth_limit = 3;
  tab_size = 2;
  max_windows = isElectron ? 50 : 15;
  default_language = isTranslateMode ? 'ach-UG' : 'en-US';
  languages = {
    'en-US': 'English',
    'fr-FR': 'French',
    'es-ES': 'EspaÃ±ol',
    'cs-CZ': 'Czech',
    'de-DE': 'German',
    'pt-BR': 'Brazilian',
    'ru-RU': 'Russian',
    'uk-UA': 'Ukrainian',
    'zh-CN': 'Chinese Simplified',
    'ja-JP': 'Japanese',
    'sr-SP': 'Serbian',
    'it-IT': 'Italian',
    'pl-PL': 'Polish',
    'ko-KR': 'Korean',
    'ro-RO': 'Romanian',
    'vi-VN': 'Vietnamese',
  };
  query_history_depth = isElectron ? 100 : 15;
  defaultTheme = 'system';
  themes = [ 'light', 'dark', 'dracula', 'system' ];
  isTranslateMode = isTranslateMode;
  isWebApp = (window as any).__ALTAIR_WEB_APP__;
  initialData = {
    url: '',
    subscriptionsEndpoint: '',
    subscriptionsProtocol: '',
    query: '',
    variables: '',
    // Force type of header, since initial value inference is wrong
    headers: (null as unknown as IDictionary),
    environments: ({} as IInitialEnvironments),
    preRequestScript: '',
    postRequestScript: '',
    instanceStorageNamespace: 'altair_',
    settings: (undefined as unknown as AltairConfigOptions['initialSettings']),
    initialSubscriptionsProvider: undefined as AltairConfigOptions['initialSubscriptionsProvider'],
    initialSubscriptionsPayload: {} as IDictionary,
    initialHttpMethod: 'POST' as HttpVerb,
    preserveState: true,
    windows: [] as AltairWindowOptions[],
  };
  constructor({
    endpointURL,
    subscriptionsEndpoint,
    subscriptionsProtocol,
    initialQuery,
    initialHeaders,
    initialEnvironments,
    initialVariables,
    initialPreRequestScript,
    initialPostRequestScript = '',
    instanceStorageNamespace,
    initialSettings,
    initialSubscriptionsProvider = WEBSOCKET_PROVIDER_ID,
    initialSubscriptionsPayload = {},
    initialHttpMethod = 'POST',
    preserveState = true,
    initialWindows = [],
  }: AltairConfigOptions = {}) {
    this.initialData.url = (window as any).__ALTAIR_ENDPOINT_URL__ || endpointURL || '';
    this.initialData.subscriptionsEndpoint = (window as any).__ALTAIR_SUBSCRIPTIONS_ENDPOINT__ || subscriptionsEndpoint || '';
    this.initialData.subscriptionsProtocol = subscriptionsProtocol;
    this.initialData.query = (window as any).__ALTAIR_INITIAL_QUERY__ || initialQuery || '';
    this.initialData.variables = (window as any).__ALTAIR_INITIAL_VARIABLES__ || initialVariables || '';
    this.initialData.headers = (window as any).__ALTAIR_INITIAL_HEADERS__ || initialHeaders || '';
    this.initialData.environments = initialEnvironments || {};
    this.initialData.preRequestScript = (window as any).__ALTAIR_INITIAL_PRE_REQUEST_SCRIPT__ || initialPreRequestScript || '';
    this.initialData.postRequestScript = initialPostRequestScript;
    this.initialData.instanceStorageNamespace = (window as any).__ALTAIR_INSTANCE_STORAGE_NAMESPACE__ || instanceStorageNamespace || 'altair_';
    this.initialData.settings = initialSettings;
    this.initialData.initialSubscriptionsProvider = initialSubscriptionsProvider;
    this.initialData.initialSubscriptionsPayload = initialSubscriptionsPayload;
    this.initialData.initialHttpMethod = initialHttpMethod;
    this.initialData.preserveState = preserveState;
    this.initialData.windows = initialWindows;
  }
}

let config = new AltairConfig();

export const setAltairConfig = (_config: AltairConfig) => {
  config = _config;
};

export const getAltairConfig = () => {
  return config;
};
(window as any).getAltairConfig = getAltairConfig;
