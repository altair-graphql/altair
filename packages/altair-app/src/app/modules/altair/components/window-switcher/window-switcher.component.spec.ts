import { expect, describe, it } from '@jest/globals';
import { WindowSwitcherComponent } from './window-switcher.component';
import { SharedModule } from '../../modules/shared/shared.module';
import { NgxTestWrapper, mount } from '../../../../../testing';
import { MockModule } from 'ng-mocks';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { AltairConfig } from 'altair-graphql-core/build/config';
import { NzContextMenuService } from 'ng-zorro-antd/dropdown';

describe('WindowSwitcherComponent', () => {
  let wrapper: NgxTestWrapper<WindowSwitcherComponent>;

  beforeEach(async () => {
    wrapper = await mount({
      component: WindowSwitcherComponent,
      imports: [MockModule(SharedModule)],
      schemas: [NO_ERRORS_SCHEMA],
      providers: [
        {
          provide: AltairConfig,
          useValue: new AltairConfig(),
        },
        {
          provide: NzContextMenuService,
          useValue: {},
        },
      ],
      propsData: {
        windows: {},
        windowIds: [],
        closedWindows: [],
        activeWindowId: '',
      },
    });
  });

  it('should create', () => {
    expect(wrapper.componentInstance).toBeTruthy();
  });

  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
