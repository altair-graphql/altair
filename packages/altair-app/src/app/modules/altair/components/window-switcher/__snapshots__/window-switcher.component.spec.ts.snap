// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WindowSwitcherComponent should render correctly 1`] = `
<app-window-switcher
  class="window-switcher__no-scrollbar"
>
  <ul
    cdkdroplist=""
    cdkdroplistorientation="horizontal"
    class="window-switcher__list window-switcher__no-scrollbar"
  >
    
    <li
      class="window-switcher__item"
    >
        
    </li>
    
  </ul>
</app-window-switcher>
`;
